src/api-service/__app__/onefuzzlib/azure/containers.py

```diff
@@ -4,81 +4,128 @@
 # Licensed under the MIT License.
 
 import datetime
+import logging
 import os
 import urllib.parse
-from enum import Enum
-from typing import Any, Dict, Optional, Union, cast
+from typing import Dict, Optional, Union, cast
 
 from azure.common import AzureHttpError, AzureMissingResourceHttpError
-from azure.storage.blob import BlobPermissions, ContainerPermissions
+from azure.storage.blob import BlobPermissions, BlockBlobService, ContainerPermissions
 from memoization import cached
+from onefuzztypes.primitives import Container
 
-from .creds import get_blob_service, get_func_storage, get_fuzz_storage
+from .storage import (
+    StorageType,
+    choose_account,
+    get_accounts,
+    get_storage_account_name_key,
+)
 
 
-class StorageType(Enum):
-    corpus = "corpus"
-    config = "config"
+@cached
+def get_blob_service(account_id: str) -> BlockBlobService:
+    logging.debug("getting blob container (account_id: %s)", account_id)
+    account_name, account_key = get_storage_account_name_key(account_id)
+    service = BlockBlobService(account_name=account_name, account_key=account_key)
+    return service
 
 
-def get_account_id_by_type(storage_type: StorageType) -> str:
-    if storage_type == StorageType.corpus:
-        account_id = get_fuzz_storage()
-    elif storage_type == StorageType.config:
-        account_id = get_func_storage()
-    else:
-        raise NotImplementedError
-    return account_id
-
-
-@cached(ttl=5)
-def get_blob_service_by_type(storage_type: StorageType) -> Any:
-    account_id = get_account_id_by_type(storage_type)
-    return get_blob_service(account_id)
+def get_service_by_container(
```

Comment:
Would it be useful to cache this, too?

Reply:
It would add difficulties relating to having to expire the cache when containers are deleted.  Caching at the account level, by `get_blob_service`, provides the majority of the benefit caching here would provide.